processing/src/main/java/org/apache/druid/segment/column/ColumnType.java

```diff
@@ -128,4 +134,80 @@ public static ColumnType ofComplex(@Nullable String complexTypeName)
   {
     return ColumnTypeFactory.getInstance().ofComplex(complexTypeName);
   }
+
+  public static ColumnType leastRestrictiveType(@Nullable ColumnType type, @Nullable ColumnType other)
```

Comment:
Comments on the method declaration:

- Method itself should be `@Nullable`? Since both inputs could be null, and then this'll return null.
- Please include javadoc about what happens if a least restrictive type can't be found. Do you get null? An exception? (From the impl, looks like you get IllegalArgumentException.)